packages/image_picker/image_picker_windows/test/image_picker_windows_test.dart

```diff
@@ -0,0 +1,104 @@
+    late MockFileSelectorPlatform mockFileSelectorPlatform;
+
+    setUp(() {
+      mockFileSelectorPlatform = MockFileSelectorPlatform();
+
+      when(mockFileSelectorPlatform.openFile(
+              acceptedTypeGroups: anyNamed('acceptedTypeGroups')))
+          .thenAnswer((_) async => null);
+
+      when(mockFileSelectorPlatform.openFiles(
+              acceptedTypeGroups: anyNamed('acceptedTypeGroups')))
+          .thenAnswer((_) async => List<XFile>.empty());
+
+      ImagePickerWindows.fileSelectorInstance = mockFileSelectorPlatform;
+    });
+
+    test('registered instance', () {
+      ImagePickerWindows.registerWith();
+      expect(ImagePickerPlatform.instance, isA<ImagePickerWindows>());
+    });
+
+    group('images', () {
+      test('pickImage passes the accepted type groups correctly', () async {
+        await plugin.pickImage(source: ImageSource.gallery);
+
+        expect(
+            verify(mockFileSelectorPlatform.openFile(
+                    acceptedTypeGroups: captureAnyNamed('acceptedTypeGroups')))
+                .captured
+                .single[0]
+                .extensions,
+            ImagePickerWindows.imageFormats);
+      });
+
+      test('getImage passes the accepted type groups correctly', () async {
+        await plugin.getImage(source: ImageSource.gallery);
+
+        expect(
+            verify(mockFileSelectorPlatform.openFile(
+                    acceptedTypeGroups: captureAnyNamed('acceptedTypeGroups')))
+                .captured
+                .single[0]
+                .extensions,
+            ImagePickerWindows.imageFormats);
+      });
+
+      test('getMultiImage passes the accepted type groups correctly', () async {
+        await plugin.getMultiImage();
+
+        expect(
+            verify(mockFileSelectorPlatform.openFiles(
+                    acceptedTypeGroups: captureAnyNamed('acceptedTypeGroups')))
+                .captured
+                .single[0]
+                .extensions,
+            ImagePickerWindows.imageFormats);
+      });
+    });
+    group('videos', () {
+      test('pickVideo passes the accepted type groups correctly', () async {
+        await plugin.pickVideo(source: ImageSource.gallery);
+
+        expect(
+            verify(mockFileSelectorPlatform.openFile(
+                    acceptedTypeGroups: captureAnyNamed('acceptedTypeGroups')))
+                .captured
+                .single[0]
+                .extensions,
+            ImagePickerWindows.videoFormats);
+      });
+
+      test('getVideo passes the accepted type groups correctly', () async {
+        await plugin.getVideo(source: ImageSource.gallery);
+
+        expect(
+            verify(mockFileSelectorPlatform.openFile(
+                    acceptedTypeGroups: captureAnyNamed('acceptedTypeGroups')))
+                .captured
+                .single[0]
+                .extensions,
+            ImagePickerWindows.videoFormats);
+      });
+    });
```

Comment:
It would be good to add tests that passing `camera` as a source throws `UnimplementedError` for all the relevant methods.

Reply:
Done.